eclass/java-pkg-simple.eclass

```diff
@@ -453,18 +565,42 @@ java-pkg-simple_src_compile() {
 			|| die "updating MANIFEST.MF failed"
 		rm -f "${T}/add-to-MANIFEST.MF" || die "cannot remove"
 	fi
+	JAVA_INTERNAL_JAR_FILES+=( ${JAVA_JAR_FILENAME} )
+}
+
+# @FUNCTION: java-pkg-simple_src_compile
+# @DESCRIPTION:
+# src_compile for simple bare source java packages. Compiles a single
+# jar via java-pkg-simple_compile_jar and calls
+# java-pkg-simple_call_ejavadoc to generate documentation if USE FLAG
+# 'doc' exists and is set.
+java-pkg-simple_src_compile() {
+	java-pkg-simple_compile_jar
+
+	# javadoc
+	if has doc ${JAVA_PKG_IUSE} && use doc; then
+		if has binary ${JAVA_PKG_IUSE} && use binary; then
+			ewarn "Generation of documentation is not supported when installing binary."
```

Comment:
Before, `src_compile` would not generate the documentation, if the `binary` use flag was active. Should we keep this, or should we still use the provided sources to generate it? How about installing the sources, when use flag `sources` is active? I think it should be handled the same as the doc generation. Either we drop both if `binary` is active, or we do both regardless.